application/app/src/main/java/com/example/buber/Views/Activities/FormUtilities/LoginFormUtils.java

```diff
@@ -0,0 +1,55 @@
+package com.example.buber.Views.Activities.FormUtilities;
+
+import android.widget.EditText;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthEmailException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LoginFormUtils {
+
+
+    public static final String FirebaseAuthEmailExceptionMSG = "There is no user record corresponding to this identifier. The user may have been deleted.";
+    public static final String FirebaseAuthPassWordExceptionCode = "ERROR_WRONG_PASSWORD";
+
+    public static boolean validateForm(EditText editEmail, EditText editPassword) {
+
+        String email = editEmail.getText().toString().trim();
+        String password = editPassword.getText().toString().trim();
+
+
+        // Referece: https://howtodoinjava.com/regex/java-regex-validate-email-address/
+        // Firebase enforces strict email formatting
+        String emailRegex = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^-]+(?:\\.[a-zA-Z0-9_!#$%&'*+/=?`{|}~^-]+)*@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$";
+        Pattern emailPattern = Pattern.compile(emailRegex);
+
+        // EMAIL
+        if (email.isEmpty()) {
+            editEmail.setError("Please provide your email");
+            return false;
+        } else {
+            editEmail.setError(null);
+        }
+        Matcher emailMatcher = emailPattern.matcher(email);
+        if (!emailMatcher.matches()) {
+            editEmail.setError("Please provide a valid email");
+            return false;
+        } else {
+            editEmail.setError(null);
+        }
+
```

Comment:
We could avoid all this by signing up using a username and appending "@buber.com " in the back end.